spec/rspec_matchers/schema_spec.rb

```diff
@@ -0,0 +1,127 @@
+require 'rails_helper'
+
+module ActiveModelSerializers
+  module Test
+    module SchemaTest
+      class MyController < ActionController::Base
+        def index
+          render json: profile
+        end
+
+        def show
+          index
+        end
+
+        def name_as_a_integer
+          profile.name = 1
+          index
+        end
+
+        def render_using_json_api
+          render json: profile, adapter: :json_api
+        end
+
+        def invalid_json_body
+          render json: ''
+        end
+
+        private
+
+        def profile
+          @profile ||= Profile.new(name: 'Name 1', description: 'Description 1', comments: 'Comments 1')
+        end
+      end
+    end
+  end
+end
+
+describe ActiveModelSerializers::Test::SchemaTest::MyController, type: :controller do
+  routes { Routes }
+
+  include ActiveModelSerializers::RSpecMatchers::Schema
```

Comment:
Something like this? 
https://gist.github.com/leonelgalan/b63d031ece0f3989185c087be3cea285

_api/lib/active_model_serializers/rspec.rb_:


```ruby
module ActiveModelSerializers
  # @api public
  # Container module for active_model_serializers specific matchers.
  module RSpecMatchers
    extend ActiveSupport::Autoload
    autoload :Schema, 'active_model_serializers/rspec_matchers/schema'
  end
end

RSpec.configure do |config|
  config.include ActiveModelSerializers::RSpecMatchers::Schema, type: :request
  config.include ActiveModelSerializers::RSpecMatchers::Schema, type: :controller
end
```